inc/functions.php

```diff
@@ -3050,30 +3050,52 @@ function random_str($length=8, $complex=false)
  */
 function format_name($username, $usergroup, $displaygroup=0)
 {
-	global $groupscache, $cache;
+	global $groupscache, $cache, $plugins;
 
-	if(!is_array($groupscache))
-	{
-		$groupscache = $cache->read("usergroups");
-	}
+	static $formattednames;
 
-	if($displaygroup != 0)
+	if(!isset($formattednames[$username]))
 	{
-		$usergroup = $displaygroup;
-	}
+		if(!is_array($groupscache))
+		{
+			$groupscache = $cache->read("usergroups");
+		}
 
-	$ugroup = $groupscache[$usergroup];
-	$format = $ugroup['namestyle'];
-	$userin = substr_count($format, "{username}");
+		if($displaygroup != 0)
+		{
+			$usergroup = $displaygroup;
+		}
 
-	if($userin == 0)
-	{
-		$format = "{username}";
-	}
+		if(isset($groupscache[$usergroup]))
+		{
+			$ugroup = $groupscache[$usergroup];
 
-	$format = stripslashes($format);
+			if(strpos($ugroup['namestyle'], "{username}") === false)
```

Comment:
Shouldn't this be `!== false`, as you use the group namestyle only if it contains the `{username}` string. The current logic only uses it if it __doesn't__ contain the `{username}` string.